prediction_market_agent/db/pinecone_handler.py

```diff
@@ -20,20 +21,32 @@
 )
 from prediction_market_agent.utils import APIKeys
 
-INDEX_NAME = "omen-markets"
+INDEX_NAME = "omen-index-text-embeddings-3-large"
 T = t.TypeVar("T")
 
 
 class PineconeHandler:
-    def __init__(self) -> None:
-        keys = APIKeys()
-        self.pc = Pinecone(api_key=keys.pinecone_api_key.get_secret_value())
-        self.index = self.pc.Index(INDEX_NAME)
+    vectorstore: VectorStore
+    pc: Pinecone
+    index: Index
+
+    def __init__(self, model: str = "text-embedding-3-large") -> None:
+        self.keys = APIKeys()
+        self.model = model
         self.embeddings = OpenAIEmbeddings(
-            api_key=keys.openai_api_key.get_secret_value()
+            api_key=self.keys.openai_api_key,
+            model=model,
         )
+        self.build_pinecone()
+        self.build_vectorstore()
+
+    def build_pinecone(self) -> None:
+        self.pc = Pinecone(api_key=self.keys.pinecone_api_key.get_secret_value())
```

Comment:
Is there a reason not to initialize self.pc, self. index, self.vectorstore in `__init__`? If no, please move it there, if yes then they should be created as `self.pc = None` and then inited here, but that will create some unnecessary type-checking troubles.

Reply:
Reason is I didn't want to have PINECONE_API_KEY as a requirement for tests.
As it's currently implemented, I can simply mock `build_pinecone` with a `return None` function and it works. See https://github.com/gnosis/prediction-market-agent/pull/307/files#diff-a3fce0cf7d663446c86cdbb35e885419e6711a20305e8a2a1630b00948f1ac78R31-R34.

Agree with `self.pc = None` and etc.